mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -87,7 +87,7 @@ impl Default for S3ClientConfig {
             auth_config: Default::default(),
             throughput_target_gbps: 10.0,
             part_size: 8 * 1024 * 1024,
-            endpoint: None,
+            endpoint_config: EndpointConfig::new("us-east-1"),
```

Comment:
feels weird still, but let's not try to address in this PR. we are planning to have some automatic region lookup in #274.

Ideal state would for this to be optional, and S3ClientConfig to contain region. Not sure how to get there yet.